framework/source/class/qx/ui/form/VirtualComboBox.js

```diff
@@ -359,7 +359,7 @@ qx.Class.define("qx.ui.form.VirtualComboBox",
       var dropdown = this.getChildControl("dropdown");
       var selection = dropdown.getSelection();
 
-      if (this.__convertValue(selection.getItem(0)) !== value)
+      if (selection.length && this.__convertValue(selection.getItem(0)) !== value)
```

Comment:
The problem is, that `__selectFirstMatch` also needs to execute its code even without an existing selection, as it has to run `dropdown.setPreselected(modelItem)` on the dropdown widget.

To solve this we have to ask for the selection not being empty OR the current value not being the current selection, only resetting the selection if there was one.

@mever I could hijack your PR and change it if you allow me to do so.

Reply:
Hi, sorry for the delay.
Be my guest :+1: